From 5ce61534c1fa70709b4448e6bc81c506bcdbb415 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 4 May 2016 10:01:35 +0300
Subject: [PATCH 0200/1345] dts: a70x0: move node status of I2C, Serial, and
 SDHCI to dts file

commit  cdf94bc1232edaf0c6dae71af42627cf652aa14a from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Move node status of I2C, Serial, and SDHCI to dts file to be
more easily to disable/enable the unit

Change-Id: I72e4185bb62cd892d6c3accc27585c0128b08895
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29486
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29572
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/marvell/armada-7040-db-router.dts     |   21 ++++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi    |    8 --------
 2 files changed, 21 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
index ab33841..0c01228 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
@@ -55,6 +55,21 @@
 
 	chosen { };
 
+	ap806 {
+		config-space@f0000000 {
+			sdhci@6e0000 {
+				status = "okay";
+			};
+			serial@512000 {
+				status = "okay";
+			};
+			i2c@511000 {
+				status = "okay";
+				clock-frequency = <100000>;
+			};
+		};
+	};
+
 	cp110-master {
 		config-space@f2000000 {
 			serial@702000 {
@@ -81,6 +96,12 @@
 			spi@700680 {
 				status = "okay";
 			};
+			i2c1: i2c@701000 {
+				status = "disabled";
+			};
+			i2c2: i2c@701100 {
+				status = "disabled";
+			};
 
 			mdio@12a200 {
 				#address-cells = <1>;
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 164027c..b4465a4 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -69,14 +69,6 @@
 			sdhci@6e0000 {
 				bus-width = <4>;
 				non-removable; /* For now, need to configure the GPIO for card detect */
-				status = "okay";
-			};
-			serial@512000 {
-				status = "okay";
-			};
-			i2c@511000 {
-				status = "okay";
-				clock-frequency = <100000>;
 			};
 		};
 	};
-- 
1.7.9.5

